MonstieWash/Assets/Scripts/MonsterAi/MonsterBrain.cs

```diff
@@ -0,0 +1,308 @@
+            var percentageStrength = ((activeMoods[i] - moodData[i].MoodLowerLimit) * 100) / (moodData[i].MoodUpperLimit - moodData[i].MoodLowerLimit);
+            float chainAmount = (percentageStrength / 100) * MoodData[i].NegativeReactionStrength * designerSanityBuff; //numbers are incredibly small *10 makes it more reasonable for designers.
+
+            //Loop through list of negative reactions in mood.
+            for (int j = 0; j < moodData[i].NegativeChainReactions.Count; j++)
+            {
+                int targetMood = accessActiveMoodIndexByMoodType(moodData[i].NegativeChainReactions[j]);
+
+                //Apply negative chain amount to each active mood that is in the list of negative reactions.
+                activeMoods[targetMood] -= chainAmount * Time.deltaTime;
+
+                if (Debug) print("Active Mood: " + moodData[i].MoodName + " negatively influenced " + moodData[targetMood].MoodName + " by amount " + chainAmount * Time.deltaTime);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Calls other function maintain limit for each mood currently active.
+    /// </summary>
+    private void maintainLimits()
+    {
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            maintainLimit(i);
+        }
+    }
+
+    /// <summary>
+    /// Checks if mood given by its current index in active list has gone over or under its limits and then corrects it back.
+    /// </summary>
+    /// <param name="moodInt"> Given index of desired mood in active moods.</param> 
+    private void maintainLimit(int moodInt)
+    {
+        //Check upper limit and set back if over.
+        if (activeMoods[moodInt] > moodData[moodInt].MoodUpperLimit) activeMoods[moodInt] = moodData[moodInt].MoodUpperLimit;
+        //Check lower limit and set back if under.
+        if (activeMoods[moodInt] < moodData[moodInt].MoodLowerLimit) activeMoods[moodInt] = moodData[moodInt].MoodLowerLimit;
+    }
+
+    /// <summary>
+    /// Takes all moodtypes and stores them in activeMood dictionary, where first value is their index in moodtypes list and the second is their current value;
+    /// </summary>
+    private void loadMoods()
+    {
+        for(int i = 0; i < moodData.Count; i++)
+        {
+            activeMoods.Add(i, moodData[i].MoodStartingPoint);
+            activeMoodNames.Add(moodData[i].MoodName,i);
+        }
+    }
+
+
+    /// <summary>
+    /// Updates ui textbox with information useful to debugging.
+    /// </summary>
+    private void updateDebugText()
+    {
+        DebugUi.text = "";
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            DebugUi.text += "MoodName: " + moodData[i].MoodName + " MoodValue: " + Mathf.FloorToInt(activeMoods[i]).ToString() + "\n" + 
+            " Mood Lower/Upper Limits: " + moodData[i].MoodLowerLimit.ToString() + "/" + moodData[i].MoodUpperLimit.ToString();
+            DebugUi.text += "\n";
+            DebugUi.text += "\n";
```

Comment:
This should be done using interpolation rather than concatenation as it's much more performant (not a super big deal since it's just a debug feature but still good practice to do all things like this with interpolation). Plus it's the convention we're using for this project.
```suggestion
            DebugUi.text += $"MoodName: {moodData[i].MoodName} 
            MoodValue: {Mathf.FloorToInt(activeMoods[i]).ToString()}\n 
            Mood Lower/Upper Limits: {moodData[i].MoodLowerLimit.ToString()}/{moodData[i].MoodUpperLimit.ToString()}\n\n";
```